staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -206,6 +211,31 @@ func GetHTTPClient(req *http.Request) string {
 	return "unknown"
 }
 
+// MaybeTruncateUserAgent truncate User-Agent if too long, otherwise return it directly.
+func MaybeTruncateUserAgent(req *http.Request) string {
```

Comment:
this doesn't seem general-purpose enough to include in a util package, I'd make it a private helper method in the handlers package for now